### PR TITLE
feat: added preview in deliveries table

### LINKED ISSUE #634: [DF-435] create a "preview" column in the deliveries table
Modify the `packages/dashboard/src/components/deliveriesTable.tsx` component by adding a preview column. This column should use an "eye" icon.

Clicking on this column should bring up a full screen preview of what the delivery looked like for users. This should reuse the preview portion of the `packages/dashboard/src/components/templateEditor.tsx` component, including the rising animation.

<sub>From [SyncLinear.com](https://synclinear.com) | [DF-435](https://linear.app/dittofeed/issue/DF-435/create-a-preview-column-in-the-deliveries-table)</sub>

### PR DESCRIPTION
**Pull request by mehulmathur16**
/claim #634
/fixes #634

https://github.com/dittofeed/dittofeed/assets/64700961/9fb47da8-7332-43f1-bf14-5cb0929047db

### CODE REVIEW

**Comment by mehulmathur16** [comment]
> Can you include the other parts of the email, like the subject, from, and to, in the preview?

Hey @maxgurewitz, I have added a preview header to email. Also I wasn't able to configure twilio for sending sms locally. Can you please check if sms previews are also working correctly, or please share a db query for manually inserting sms deliveries.

https://github.com/dittofeed/dittofeed/assets/64700961/485a0f38-75a2-4b25-925f-f3cf9a464191

File: packages/dashboard/src/components/deliveriesTable.tsx [hunk 1]
```diff
@@ -40,6 +56,10 @@ interface TableItem {
   templateId?: string;
   templateName?: string;
   channel: ChannelType;
+  body: string | null;
```
Comment (maxgurewitz) [inline packages/dashboard/src/components/deliveriesTable.tsx:59]: Yah, it does look like you've written this code to only work with email deliveries. I'll help you figure out how to test it with SMS.

File: packages/dashboard/src/components/deliveriesTable.tsx [hunk 2]
```diff
@@ -48,6 +68,48 @@ interface DeliveriesState {
...
+    from: "",
+    replyTo: "",
+    to: "",
+    body: "",
+    show: false,
+    channelType: "",
+  };
+};
+
+const BodyBox = styled(Box, {
```
Comment (maxgurewitz) [inline packages/dashboard/src/components/deliveriesTable.tsx:93]: I believe you've copy and pasted this code. Maybe you can pull it out into a component to share instead?

File: packages/dashboard/src/components/deliveriesTable.tsx [hunk 3]
```diff
@@ -345,120 +440,321 @@ export function DeliveriesTable({
...
-        rows={rows}
-        loading={paginationRequest.type === CompletionStatus.InProgress}
-        columns={[
-          {
-            field: "userId",
-            headerName: "User ID",
-            renderCell: ({ row }: GridRenderCellParams<TableItem>) => {
-              const href = `/users/${row.userId}`;
-              return <LinkCell href={href} title={row.userId} />;
+  const renderEmailPreviewBody = (body: string) => {
```
Comment (maxgurewitz) [inline packages/dashboard/src/components/deliveriesTable.tsx:443]: It looks like you've copy and pasted much of this code from `packages/dashboard/src/components/messages/emailEditor.tsx` can you share the code instead?

**Review by maxgurewitz — CHANGES_REQUESTED** [review]

**Comment by maxgurewitz** [comment]
@mehulmathur16 I've added these two issues which should make testing SMS easier, by selecting the test email provider.

https://github.com/dittofeed/dittofeed/issues/650
https://github.com/dittofeed/dittofeed/issues/651

**Comment by mehulmathur16** [comment]
@maxgurewitz The below video shows that the preview functionality is working fine for both email and sms, in template editor page as well as delivery page.


https://github.com/dittofeed/dittofeed/assets/64700961/d25db12c-c148-4f9c-82de-bf1e87d2bad1

File: packages/dashboard/src/components/deliveriesTable.tsx (hunk 1 above)
Comment (mehulmathur16) [reply packages/dashboard/src/components/deliveriesTable.tsx:55]: This is working fine for sms. I've tested it.

File: packages/dashboard/src/components/emailPreviewHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import { SxProps, TextField, Theme, useTheme } from "@mui/material";
+import escapeHTML from "escape-html";
+import React from "react";
+
+interface EmailPreviewHeaderProps {
```
Comment (maxgurewitz) [inline packages/dashboard/src/components/emailPreviewHeader.tsx:5]: Awesome, glad that you pulled this out! :)

**Review by maxgurewitz — APPROVED** [review]
LGTM :)

File: packages/dashboard/src/components/emailPreviewHeader.tsx (hunk 4 above)
Comment (mehulmathur16) [reply packages/dashboard/src/components/emailPreviewHeader.tsx:5]: Thank you Max 😄

**Pull request merged**